src/ts/localization/i18n.ts

```diff
@@ -0,0 +1,182 @@
+  'settings.subtitles.font.family.casual': string;
+  'settings.subtitles.font.family.cursive': string;
+  'settings.subtitles.font.family.smallcapital': string;
+  'settings.subtitles.window.color': string;
+  'settings.subtitles.window.opacity': string;
+  'colors.white': string;
+  'colors.black': string;
+  'colors.red': string;
+  'colors.green': string;
+  'colors.blue': string;
+  'colors.cyan': string;
+  'colors.yellow': string;
+  'colors.magenta': string;
+  'opacity.100': string;
+  'opacity.75': string;
+  'opacity.50': string;
+  'opacity.25': string;
+  'opacity.0': string;
+  'settings': string;
+  'ads.remainingTime': string;
+  'pictureInPicture': string;
+  'appleAirplay': string;
+  'googleCast': string;
+  'vr': string;
+  'off': string;
+  'auto': string;
+  'back': string;
+  'reset': string;
+  'replay': string;
+  'normal': string;
+  'default': string;
+  'open': string;
+  'fullscreen': string;
+  'speed': string;
+  'playPause': string;
+}
+
+export type CustomVocabulary<V> = V & Partial<Vocabulary>;
+type StringKeyMap = {[key: string]: string};
+
+export interface BitmovinPlayerUiTranslations {
+  [key: string]: CustomVocabulary<StringKeyMap>;
+}
+
+
+
+export default class I18n {
+  private language: string;
+  private vocabulary: CustomVocabulary<StringKeyMap>;
+
+  constructor(config: LocalizationConfig) {
+    this.setConfig(config);
+  }
+
+  public setConfig(config: LocalizationConfig) {
+    const translations = this.mergeTranslationsWithDefaultTranslations(config.translations);
+    this.initializeLanguage(config.language, config.disableBrowserLanguageDetection, translations);
+    const fallbackLanguages = this. getConfiguredFallbackLanguages(translations, config.fallbackLanguages);
+    this.initializeVocabulary(translations, fallbackLanguages);
+  }
+
+  private containsKey(obj: object, key: string) {
+    return Object.keys(obj).indexOf(key) !== -1;
+  }
+
+  private mergeTranslationsWithDefaultTranslations(translations: BitmovinPlayerUiTranslations) {
+    const rawTranslations: BitmovinPlayerUiTranslations = { ...defaultTranslations, ...translations};
+    return Object.keys(rawTranslations).reduce((acc, key) => {
+      let translation: CustomVocabulary<StringKeyMap> = rawTranslations[key as string];
+      if (this.containsKey(defaultTranslations, key) && this.containsKey(translations, key)) {
+        translation = {...defaultTranslations[key], ...translations[key]};
+      }
+      return { ...acc, [key]: translation };
+    }, {} as BitmovinPlayerUiTranslations);
+  }
+
+  private initializeLanguage(
+    language: string,
+    disableBrowserLanguageDetection: boolean,
+    translations: BitmovinPlayerUiTranslations,
+  ) {
+    const shouldDetectLanguage = !Boolean(disableBrowserLanguageDetection);
+
+    if (shouldDetectLanguage) {
+      let userLanguage = (window.navigator.language);
+      userLanguage = userLanguage.slice(0, 2);
```

Comment:
Language names are not limited to 2 chars, what if it's called `de-at`?

Reply:
it is true, I have been thinking about it after your comment about the multi-language & fallback languages. 
the problem I encountered is there are too many languages and too many variations. and basically if you are localizing a website you localize it for `de` not `de-at` or `en` not `en-us` (theoretically)  for example, for english we have:
```
| English - Australia | en | en-au |
| English - Belize | en | en-bz |
| English - Canada | en | en-ca |
| English - Caribbean | en | en-cb |
| English - Great Britain | en | en-gb | 
| English - India | en | en-in |
| English - Ireland | en | en-ie |
| English - Jamaica | en | en-jm |
| English - New Zealand | en | en-nz |
| English - Philippines | en | en-ph |
| English - Southern Africa | en | en-za |
| English - Trinidad | en | en-tt |
| English - United States | en | en-us |
| English - Zimbabwe | en |
```
also depending on the browser return value can be `en-AU` or `en-au` or just `en`...
should user add the same translation for all or should we have a map containing all of the language maps? 
that is why I taught using just the first `2` chars of the language is the safest way to go and for each [language variation first 2 is same anyways](https://www.science.co.il/language/Locale-codes.php)... 

this supporting the de-at would also create confusion (atleast for me )

user-defined-vocabularies: 'de-DE', 'en-US', 'it'
default-language: 'en-US'
browser-language: 'de-AT'
fallbackLangs in order: 'it', 'de-DE' 
in this case we will not set language to `de-DE` because browser finds `de-AT` and in our vocabulary there is no such thing. so the language will be set to `en-US` even tough the `de` is available...

Reply:
so my suggestion is to standardize this somehow. and `recommend` user to use those standards to get expected results`

Reply:
The "shortnames" are already some kind of standard. Simply check for the full string, e.g. `en-us`, and fallback to `en` otherwise.